docarray/array/storage/memory/getsetdel.py

```diff
@@ -14,23 +14,22 @@ class GetSetDelMixin(BaseGetSetDelMixin):
 
     def _del_docs_by_mask(self, mask: Sequence[bool]):
         self._data = list(itertools.compress(self._data, (not _i for _i in mask)))
-        self._rebuild_id2offset()
+        self._needs_id2offset_rebuild = True
 
     def _del_all_docs(self):
         self._data.clear()
         self._id2offset.clear()
 
     def _del_docs_by_slice(self, _slice: slice):
         del self._data[_slice]
-        self._rebuild_id2offset()
+        self._needs_id2offset_rebuild = True
```

Comment:
change the `_needs_id2offset_rebuild` to a decorator?

Reply:
I changed it in the last commit, could you review ?